src/common/actions/snaps.js

```diff
@@ -42,6 +42,20 @@ export function fetchUserSnaps(owner) {
   };
 }
 
+function shouldFetchUserSnaps(state) {
+  const snaps = state.snaps;
+
+  return !snaps.isFetching;
+}
+
+export function fetchUserSnapsIfNeeded(owner) {
+  return function(dispatch, getState) {
+    if (shouldFetchUserSnaps(getState())) {
```

Comment:
No sure if `auth` is a good example but for example here: https://github.com/canonical-ols/build.snapcraft.io/blob/master/src/common/components/repositories-list/index.js#L23-L37

All I mean is that by this time we had all the logic and conditional dispatching in components that were using the data.

But if it's a pattern from redux docs then I'm less scared that we try to do something unusual.
And, as mentioned above I think in theory we would need that for all the fetch actions.
Maybe it could even be the default - so instead of `fetchSnapsIfNeeded` being conditional simply `fetchSnaps` should fetch only if we are not fetching the data already.

But that's something we can think about later when we start to use this pattern more.